pkg/metric/metric.go

```
@@ -185,3 +187,7 @@ func (pmr *prometheusMetricsRegistry) GetHTTPMetricsInstrumentationsMiddlewareFo
 	}
 	return httpInstrumentation.Middleware, nil
 }
+
+func (pmr *prometheusMetricsRegistry) RegisterCollector(ctx context.Context, collectors prometheus.Collector) {
```

nit: let's name this `MustRegisterCollector`. The `Must` is a Go-ism to indicate - this will panic rather than return an error if it fails.

Arguably we should have a `RegisterCollector` that calls the non-must version of the func and bubbles up the error.

---

Yes, I had wondered the same on using `Register` instead but followed the MustRegister pattern with the default collectors on the registry - https://github.com/hyperledger/firefly-common/pull/155/files#diff-8b48568f0ecef3176c9d0ccd9ace7f9c73753c2a0ad6fd03f99c8f93ac186d73R98